### PR TITLE
[PARKED] Add provisioning of Profile to Openconext-deploy

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket 106202952](https://www.pivotaltracker.com/story/show/106202952)

### CODE REVIEW

**Comment by arothuis** [comment]
@rjkip, @DRvanR 

There are some things in this PR that are useful for when Profile's dev provisioning is included in OpenConext-deploy. Should they be kept or added later?

**Comment by DRvanR** [comment]
imo, removed and added later, perhaps store in a different branch for the time being...

File: inventory/acc [hunk 1]
```diff
@@ -12,6 +12,10 @@ acc62.coin.surf.net
 acc41.coin.surf.net
 acc42.coin.surf.net
 
+[php56-apps]
+acc41.coin.surf.net
+acc42.coin.surf.net
```
Comment (DRvanR) [inline inventory/acc]: This cannot occur as with the current setup running two different php versions on the same machine is not supported. Which machine(s) this should be should be requested from SURFnet (ping @thijskh @quartje) or, if now unkown, should be removed completely until known.

File: inventory/diy [hunk 2]
```diff
@@ -13,6 +13,9 @@ diy01
 [php-apps]
 diy01
 
+[php56-apps]
+diy01
```
Comment (DRvanR) [inline inventory/diy]: same here

File: inventory/diy-test [hunk 3]
```diff
@@ -13,6 +13,9 @@ diy11
 [php-apps]
 diy11
 
+[php56-apps]
+diy11
```
Comment (DRvanR) [inline inventory/diy-test]: and here :wink:

File: inventory/vm [hunk 4]
```diff
@@ -21,6 +21,9 @@ java-apps
 [php-apps]
 192.168.66.79 ansible_ssh_private_key_file=./.vagrant/machines/apps/virtualbox/private_key
 
+[php56-apps]
+192.168.66.79 ansible_ssh_private_key_file=./.vagrant/machines/apps/virtualbox/private_key
```
Comment (DRvanR) [inline inventory/vm]: not possible :stuck_out_tongue:

File: roles/php56/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- name: Install PHP extensions
+  yum: name={{ item }} state=present enablerepo=remi
+  with_items:
+        - php-cli
+        - php-bcmath
+        - php-mbstring
+        - php-mcrypt
+        - php-pdo
+        - php-mysql
+        - php-gmp
```
Comment (DRvanR) [inline roles/php56/tasks/main.yml]: `gmp` should not be needed, `bcmath`, `pdo` and `mysql` are not needed now, may be in the future.

File: roles/php56/templates/apc.ini.j2 [hunk 6]
```diff
@@ -0,0 +1,3 @@
+extension=apc.so
+apc.enabled=1
+apc.shm_size=256M
```
Comment (DRvanR) [inline roles/php56/templates/apc.ini.j2]: but much it seems :wink:

File: roles/php56/files/httpd.php.conf [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+# Add index.php to the list of files that will be served as directory
+# indexes.
+#
+DirectoryIndex index.php
+
+#
+# Uncomment the following line to allow PHP to pretty-print .phps
+# files as PHP source code:
+#
+#AddType application/x-httpd-php-source .phps
```
Comment (DRvanR) [inline roles/php56/files/httpd.php.conf]: just remove this

File: roles/php56/files/httpd.php.conf [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+# Cause the PHP interpreter to handle files with a .php extension.
+#
+AddHandler php5-script .php
+AddType text/html .php
+
+#
+# Add index.php to the list of files that will be served as directory
+# indexes.
+#
+DirectoryIndex index.php
```
Comment (DRvanR) [inline roles/php56/files/httpd.php.conf]: eeehr, please no :stuck_out_tongue:

File: provision-profile.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  gather_facts: no
+  sudo: true
+  roles:
+    - common
+    - tls
+  handlers:
+    - include: roles/httpd/handlers/main.yml
+    - include: roles/nginx/handlers/main.yml
+
+- hosts: loadbalancer
```
Comment (DRvanR) [inline provision-profile.yml]: does profile really need a loadbalancer, or is that something that the general loadbalancer can fix?

File: roles/profile/defaults/main.yml [hunk 10]
```diff
@@ -0,0 +1,30 @@
+#---
+## Version of Profile that is installable by this role
+profile_version: 0.0.0
```
Comment (DRvanR) [inline roles/profile/defaults/main.yml]: ???

File: roles/profile/defaults/main.yml [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+profile_version_dir: "{{ profile_version | replace('/', '-') }}"
+profile_branch_dir: "{{ openconext_builds_dir }}/OpenConext-profile-{{ profile_branch | replace('/', '-') }}"
+profile_release_dir: "{{ openconext_releases_dir }}/OpenConext-profile-{{ profile_version_dir }}"
+profile_build_path: "{{ openconext_builds_dir }}/OpenConext-profile-{{ profile_version_dir }}.tar.gz"
+profile_download_url: "https://github.com/OpenConext/OpenConext-profile/releases/download/{{ profile_version }}/OpenConext-profile-{{ profile_version_dir }}.tar.gz"
+profile_current_release_symlink: "{{ openconext_releases_dir }}/OpenConext-profile"
+
+profile_domain: profile.{{ base_domain }}
+
+profile_default_locale: en
```
Comment (DRvanR) [inline roles/profile/defaults/main.yml]: perhaps add some documentation as to what these variables are and why they should be configured and how.

File: roles/profile/files/app_dev.php [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (DRvanR) [inline roles/profile/files/app_dev.php]: if we don't need it, can't this go?

File: roles/profile/templates/profile.conf.j2 [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    # We support only GET/POST/HEAD
+    RewriteCond %{REQUEST_METHOD} !^(POST|GET|HEAD)$
+    RewriteRule .* - [R=405,L]
+    # If the requested url does not map to a file or directory, then forward it to index.php/URL.
+    # Note that the requested URL MUST be appended because Corto uses the PATH_INFO server variable
+    RewriteCond %{DOCUMENT_ROOT}%{REQUEST_FILENAME} !-f
+    RewriteCond %{DOCUMENT_ROOT}%{REQUEST_FILENAME} !-d
+    RewriteRule ^(.*)$ /app.php/$1 [L] # Send the query string to app.php
+    # Requests to the domain (no query string)
+    RewriteRule ^$ /app.php/ [L]
```
Comment (DRvanR) [inline roles/profile/templates/profile.conf.j2]: as discussed, keep what is needed, rewrite the rest as per the [optimized configuration for Symfony][1]

[1]: http://symfony.com/doc/current/cookbook/configuration/web_server_configuration.html#apache-with-mod-php-php-cgi

File: roles/profile/templates/profile_deploy.yml.j2 [hunk 14]
```diff
@@ -0,0 +1,3 @@
+---
+cache_path: {{ profile_symfony_cache_path }}
```
Comment (DRvanR) [inline roles/profile/templates/profile_deploy.yml.j2]: might want to rename the file (and thus the name of the generated file) to `profile.yml` or something like that (akin to `/etc/openconext/engineblock.ini`)

File: roles/rsyslog/tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+---
+- name: Install rsyslog
+  yum: pkg=rsyslog state=latest
+
+- name: Configure rsyslog to allow linelength of 10K
+  lineinfile: dest=/etc/rsyslog.conf insertbefore="BOF" line="$MaxMessageSize 10k"
```
Comment (DRvanR) [inline roles/rsyslog/tasks/main.yml]: has there been any discussion with regards to handlers/filters?

**Comment by thijskh** [comment]
Due to a large &Delta; caused by the many refactorings on the `engineblock5-centos7` branch, this has many conflicts now. I extracted the new `roles/profile/` and will integrate it in a new `feature/profile-provisioning2` branch where development on this will continue.

**Pull request closed without merging**